filesystem/Upload.php

```diff
@@ -208,6 +208,7 @@ public function load($tmpFile, $folderPath = false) {
 			$this->file->ParentID = $parentFolder ? $parentFolder->ID : 0;
 			// This is to prevent it from trying to rename the file
 			$this->file->Name = basename($relativeFilePath);
+			$this->file->Title = preg_replace("/{$fileSuffix}$/", '', $tmpFile['name']);
```

Comment:
`$fileSuffix` needs to be correctly escaped for use in a regular expression. I'm not sure a regular expression is even needed here.